hdf5: make possible to use the v1.10 API

LeSuisse opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

hdf5 v1.12 has introduced [breaking changes to the API](https://github.com/HDFGroup/hdf5/issues/2) and [some dependent packages are not yet compatible with it](https://github.com/NixOS/nixpkgs/pull/118312#issuecomment-813000589). This contribution adds a way to build the hdf5 library with the v1.10 API.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

LeSuisse committed:
hdf5: make possible to use the v1.10 API

LeSuisse committed:
vigra: use the HDF5 v1.10 API

vigra is not yet compatible with the v1.12 API [0][1].

[0] https://github.com/ukoethe/vigra/issues/476
[1] https://github.com/ukoethe/vigra/issues/476

LeSuisse commented:
@FRidh you mentioned an issue with `pandas` but I was not able to find the failure on Hydra, what did I miss?

FRidh commented:
This needs to target `staging-next`.

pandas segfaults in a tables test. tables is the pytables package that uses hdf5. https://hydra.nixos.org/build/140676472

LeSuisse commented:
It looks like this solves the issue for vigra but pandas still segfault :/

r-rmcgibbo commented:
Result of `nixpkgs-review pr 118454` at 6ef6455f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>36 packages marked as broken and skipped:</summary>
  <ul>
    <li>mlflow-server</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.dftfit</li>
    <li>python38Packages.geopandas</li>
    <li>python38Packages.lammps-cython</li>
    <li>python38Packages.mlflow</li>
    <li>python38Packages.mlxtend</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>48 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/1299d8b72c14370144cdb2d5be7d42ce">python39Packages.Quandl</a> (python39Packages.quandl)</li>
    <li><a href="https://gist.github.com/a29c517c9ad49443e03b0bd5407cf523">python39Packages.acoustics</a></li>
    <li><a href="https://gist.github.com/fa071dfac7f97ec906058e5a1ab1a1d7">python39Packages.aioinflux</a></li>
    <li><a href="https://gist.github.com/c41e0b4107fb5c13ab1bbe9011c58d43">python39Packages.altair</a></li>
    <li><a href="https://gist.github.com/71ea6eaefa1f0d80d36381b0356b8d7f">python39Packages.apptools</a></li>
    <li><a href="https://gist.github.com/f9a048a43d44805bdc569bf6913d0ddc">python39Packages.asdf</a></li>
    <li><a href="https://gist.github.com/d03841fd475cd90904d6d4db5a2d16c0">python39Packages.bkcharts</a></li>
    <li><a href="https://gist.github.com/0f6cecc4473097edaaabf29a69ac8f58">python39Packages.csvs-to-sqlite</a></li>
    <li><a href="https://gist.github.com/530b073f4b0f07c52e7d3acd7ad4fb4d">python39Packages.dask</a></li>
    <li><a href="https://gist.github.com/5d0a0f55052fa4f7c9313f7d663d2ae5">python39Packages.drms</a></li>
    <li><a href="https://gist.github.com/b5150906f594d6075626e40c6e6fd484">python39Packages.envisage</a></li>
    <li><a href="https://gist.github.com/d67758375fc41e157fb13ab5ffe8b937">python39Packages.flammkuchen</a></li>
    <li><a href="https://gist.github.com/a9d4d6485d8803591effc321cece1637">python39Packages.glymur</a></li>
    <li><a href="https://gist.github.com/ef7b5b4c60347ebe49c621fca5756a18">python39Packages.holoviews</a></li>
    <li><a href="https://gist.github.com/d39f2bd5e28875c2600a069c3300450f">python39Packages.huggingface-hub</a></li>
    <li><a href="https://gist.github.com/e2b171add53f941217eabb4dffb93ded">python39Packages.hvplot</a></li>
    <li><a href="https://gist.github.com/9e8d0b9c21a569d17b4e7825c0c41927">python39Packages.influxdb</a></li>
    <li><a href="https://gist.github.com/ac943552e693b0dfd4cedfc6d615b0e2">python39Packages.mesa</a></li>
    <li><a href="https://gist.github.com/810cce62d686076da5d53e2f4cb4b42e">python39Packages.mplfinance</a></li>
    <li><a href="https://gist.github.com/e6beb5171e4f269cd373c16194e9bca7">python39Packages.osmpythontools</a></li>
    <li><a href="https://gist.github.com/c13d2fd18680f84b83e7ea6b760b708a">python39Packages.pandas</a></li>
    <li><a href="https://gist.github.com/1810d43b5cec4364060b287aa598d6f0">python39Packages.pandas-datareader</a></li>
    <li><a href="https://gist.github.com/db6396e3b079e781324577652afb3738">python39Packages.panel</a></li>
    <li><a href="https://gist.github.com/b82b1ae1147bc0494cab6bfec1c213ee">python39Packages.parfive</a></li>
    <li><a href="https://gist.github.com/955d748770d9dd8c63d85390f092cdb2">python39Packages.partd</a></li>
    <li><a href="https://gist.github.com/c78f1c2e25dc2dab3338bea7dd6d3663">python39Packages.ppscore</a></li>
    <li><a href="https://gist.github.com/a61d2420125795c9709cfa58e9fd08b4">python39Packages.pvlib</a></li>
    <li><a href="https://gist.github.com/d1f067613c4f424cbe61af9881d15ee0">python39Packages.pymatgen</a></li>
    <li><a href="https://gist.github.com/4953c0610750d9ba00c1c782fc062057">python39Packages.pytest-astropy</a></li>
    <li><a href="https://gist.github.com/8418edcb5b48b21ea8e447a18b60a462">python39Packages.pytest-astropy-header</a></li>
    <li><a href="https://gist.github.com/a4e630d28152f41e004ed343db67f5a3">python39Packages.pytrends</a></li>
    <li><a href="https://gist.github.com/6af743e77fc4dac50d38cec23b5d0f92">python39Packages.pywick</a></li>
    <li><a href="https://gist.github.com/63a27aa006bb04e232e8d7921d1e4b45">python39Packages.rapidfuzz</a></li>
    <li><a href="https://gist.github.com/26a87c0e82f204c8c484448b1774e6c0">python39Packages.sagemaker</a></li>
    <li><a href="https://gist.github.com/4b0a4dd96f23d783d0c25249e0ffd830">python39Packages.scikit-bio</a></li>
    <li><a href="https://gist.github.com/c30bc2317838dddb4bb713fe8d43e7d9">python39Packages.scikitimage</a></li>
    <li><a href="https://gist.github.com/c404010cdd61b75bc615e1671d9f966a">python39Packages.seaborn</a></li>
    <li><a href="https://gist.github.com/1bffcebd669d34a8b9b2bb4fd354dd30">python39Packages.slicer</a></li>
    <li><a href="https://gist.github.com/a891c5cda0db80667466db5ee274fdac">python39Packages.spyder-kernels_0_5</a></li>
    <li><a href="https://gist.github.com/ea650bc581502f2d6acb2537fda826b8">python39Packages.tablib</a></li>
    <li><a href="https://gist.github.com/f1cc8e4fc13bd7f87d4ce2fa07cdd319">python39Packages.test-tube</a></li>
    <li><a href="https://gist.github.com/971935850bb8cf021171b121efcbc5fc">python39Packages.tqdm</a></li>
    <li><a href="https://gist.github.com/56ff817b7f7341df66889ecb762d8c02">python39Packages.traittypes</a></li>
    <li><a href="https://gist.github.com/34dfdd854b5b50043b14cf9c984eee30">python39Packages.vega</a></li>
    <li><a href="https://gist.github.com/e28ce4d046af3561bad221ef357ba513">python39Packages.vega_datasets</a></li>
    <li><a href="https://gist.github.com/e1f9f5749df2b7467320fbb50276edb5">python39Packages.vidstab</a></li>
    <li><a href="https://gist.github.com/b60862d6284426ded1fbbedfd8554f01">python39Packages.xarray</a></li>
    <li><a href="https://gist.github.com/74de419c1e51dafa4598a8b3a721dcd3">python39Packages.yfinance</a></li>
  </ul>
</details>
<details>
  <summary>325 packages skipped due to time constraints:</summary>
  <ul>
    <li>aiodnsbrute</li>
    <li>apache-airflow</li>
    <li>backblaze-b2</li>
    <li>check_systemd</li>
    <li>checkov</li>
    <li>conan</li>
    <li>cq-editor</li>
    <li>csvs-to-sqlite (python38Packages.csvs-to-sqlite)</li>
    <li>deja-dup</li>
    <li>digikam</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>python39Packages.tables</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

FRidh commented:
> It looks like this solves the issue for vigra but pandas still segfault :/

Good vigra is fixed. Pandas could have another cause then, or it is the update. To check, it is worthwhile building pandas against the old hdf5.

LeSuisse commented:
> Good vigra is fixed. Pandas could have another cause then, or it is the update itself despite using the old api. To check, it is worthwhile building pandas against the old hdf5.

It tried to revert the upgrade and build pandas and the segfault does not happen. 

I'm trying to see what's wrong with the upgrade.

FRidh commented:
In case we won't figure out (within a day or two, so staging-next can proceed), the alternative is to keep 1.11 in just for pandas.

SuperSandro2000 commented:
> @FRidh you mentioned an issue with `pandas` but I was not able to find the failure on Hydra, what did I miss?

Fixed that with https://github.com/NixOS/nixpkgs/pull/118512

LeSuisse committed:
hdf5_1_10: init at 1.10.7

Introduce hdf5_1_10 so it can be used by packages encountering issues
hdf5 1.12.0.

LeSuisse committed:
python3Packages.tables: use hdf5_1_10 instead of hdf5

Without that, the test suite of python3Packages.pandas segfault.

LeSuisse commented:
I have not made significant progress on this topic and I'm not sure I will in a timely manner. Maybe someone with more experience of the Python ecosystem/pandas could take a look and have more success than me.

In the meantime to unblock the situation I have updated the PR to introduce `hdf5_1_10` so it can be used with `python3Packages.tables` to avoid segfault in the test suite of `python3Packages.pandas`.

SuperSandro2000 commented on pkgs/tools/misc/hdf5/1.10.nix:
```diff
@@ -0,0 +1,52 @@
+
+  nativeBuildInputs = [ removeReferencesTo ];
+
+  propagatedBuildInputs = []
+    ++ optional (zlib != null) zlib;
```
```suggestion
  propagatedBuildInputs = optionals (zlib != null) [ zlib ];
```

SuperSandro2000 commented on pkgs/tools/misc/hdf5/1.10.nix:
```diff
@@ -0,0 +1,52 @@
+  propagatedBuildInputs = []
+    ++ optional (zlib != null) zlib;
+
+  configureFlags = []
+    ++ optional enableShared "--enable-shared";
```
```suggestion
  configureFlags = optionals enableShared [ "--enable-shared" ];
```

SuperSandro2000 commented on pkgs/development/python-modules/pandas/default.nix:
```diff
@@ -110,6 +110,12 @@ buildPythonPackage rec {
+  disabledTestPaths = [
+    # some test in this file either crashes python or hangs for hours
+    # TODO: figure out why and how to fix it
+    "tests/io/pytables/test_pytables_missing.py"
```
Is this still required with hdf5 1.10?

LeSuisse replied:
You are right it's not needed :+1: 
It was a leftover of my tests :disappointed:

SuperSandro2000 replied:
Nothing to worry about. Sometimes you just need a second pair of eyes.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118454` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
37 packages marked as broken and skipped:
</summary>
<ul>
<li>mlflow-server</li>
<li>python38Packages.arviz</li>
<li>python38Packages.clifford</li>
<li>python38Packages.dask-jobqueue</li>
<li>python38Packages.dask-xgboost</li>
<li>python38Packages.dftfit</li>
<li>python38Packages.geopandas</li>
<li>python38Packages.lammps-cython</li>
<li>python38Packages.mlflow</li>
<li>python38Packages.mlxtend</li>
<li>python38Packages.optuna</li>
<li>python38Packages.osmnx</li>
<li>python38Packages.pymatgen-lammps</li>
<li>python38Packages.pymc3</li>
<li>python38Packages.rl-coach</li>
<li>python38Packages.shap</li>
<li>python38Packages.skorch</li>
<li>python38Packages.textacy</li>
<li>python38Packages.tiledb</li>
<li>python38Packages.trackpy</li>
<li>python39Packages.dask-jobqueue</li>
<li>python39Packages.dask-xgboost</li>
<li>python39Packages.geopandas</li>
<li>python39Packages.google-cloud-automl</li>
<li>python39Packages.google-cloud-bigquery</li>
<li>python39Packages.google-cloud-monitoring</li>
<li>python39Packages.lammps-cython</li>
<li>python39Packages.mlflow</li>
<li>python39Packages.mlxtend</li>
<li>python39Packages.optuna</li>
<li>python39Packages.osmnx</li>
<li>python39Packages.pymatgen-lammps</li>
<li>python39Packages.rl-coach</li>
<li>python39Packages.skorch</li>
<li>python39Packages.textacy</li>
<li>python39Packages.tiledb</li>
<li>python39Packages.trackpy</li>
</ul>
</details>
<details>

  <summary>18 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>apache-airflow: log was empty</li>
<li>python38Packages.cnvkit: log was empty</li>
<li>python38Packages.dask-glm: log was empty</li>
<li>python38Packages.dask-ml: log was empty</li>
<li>python38Packages.imgaug: log was empty</li>
<li>python38Packages.nilearn: log was empty</li>
<li>python38Packages.pywick: log was empty</li>
<li>python38Packages.tablib: log was empty</li>
<li>python38Packages.transformers: log was empty</li>
<li>python38Packages.xgboost: log was empty</li>
<li>python39Packages.cnvkit: log was empty</li>
<li>python39Packages.dask-glm: log was empty</li>
<li>python39Packages.imgaug: log was empty</li>
<li>python39Packages.nilearn: log was empty</li>
<li>python39Packages.pywick: log was empty</li>
<li>python39Packages.tablib: log was empty</li>
<li>python39Packages.transformers: log was empty</li>
<li>python39Packages.xgboost: log was empty</li>
  </ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>python38Packages.imbalanced-learn: log was empty</li>
  </ul>
</details>
<details>
<summary>
337 packages built:
</summary>
<ul>
<li>aiodnsbrute</li>
<li>backblaze-b2</li>
<li>check_systemd</li>
<li>checkov</li>
<li>conan</li>
<li>cq-editor</li>
<li>csvs-to-sqlite (python38Packages.csvs-to-sqlite)</li>
<li>deja-dup</li>
<li>dpt-rp1-py</li>
<li>duplicity</li>
<li>duply</li>
<li>enblend-enfuse</li>
<li>evdevremapkeys</li>
<li>evillimiter</li>
<li>ffmpeg-normalize</li>
<li>gdown (python38Packages.gdown)</li>
<li>gnome-keysign</li>
<li>graph-cli</li>
<li>hdf5_1_10</li>
<li>home-assistant</li>
<li>hugin</li>
<li>jellyfin-mpv-shim</li>
<li>kaggle (python38Packages.kaggle)</li>
<li>keepkey_agent (python38Packages.keepkey_agent)</li>
<li>ldeep</li>
<li>ledger_agent (python38Packages.ledger_agent)</li>
<li>lieer</li>
<li>luigi</li>
<li>magic-wormhole (python38Packages.magic-wormhole)</li>
<li>manim</li>
<li>mathlibtools (python38Packages.mathlibtools)</li>
<li>mnemosyne</li>
<li>noto-fonts-emoji</li>
<li>nrfutil</li>
<li>ocrmypdf</li>
<li>papis (python38Packages.papis)</li>
<li>past-time</li>
<li>plex-mpv-shim</li>
<li>python38Packages.Quandl (python38Packages.quandl)</li>
<li>python38Packages.acoustics</li>
<li>python38Packages.afdko</li>
<li>python38Packages.aioinflux</li>
<li>python38Packages.aiosmb</li>
<li>python38Packages.altair</li>
<li>python38Packages.ansible-kernel</li>
<li>python38Packages.ansible-runner</li>
<li>python38Packages.aplpy</li>
<li>python38Packages.apptools</li>
<li>python38Packages.asdf</li>
<li>python38Packages.atomman</li>
<li>python38Packages.b2sdk</li>
<li>python38Packages.batchgenerators</li>
<li>python38Packages.bkcharts</li>
<li>python38Packages.caffe</li>
<li>python38Packages.catboost</li>
<li>python38Packages.cirq</li>
<li>python38Packages.cufflinks</li>
<li>python38Packages.dask</li>
<li>python38Packages.dask-gateway</li>
<li>python38Packages.dask-image</li>
<li>python38Packages.dask-mpi</li>
<li>python38Packages.datasets</li>
<li>python38Packages.datashader</li>
<li>python38Packages.dipy</li>
<li>python38Packages.distributed</li>
<li>python38Packages.dnachisel</li>
<li>python38Packages.drms</li>
<li>python38Packages.duckdb</li>
<li>python38Packages.fastparquet</li>
<li>python38Packages.flammkuchen</li>
<li>python38Packages.genome-collector</li>
<li>python38Packages.glymur</li>
<li>python38Packages.google-cloud-automl</li>
<li>python38Packages.google-cloud-bigquery</li>
<li>python38Packages.google-cloud-monitoring</li>
<li>python38Packages.gradient</li>
<li>python38Packages.gradient-utils</li>
<li>python38Packages.graphtage</li>
<li>python38Packages.graspologic</li>
<li>python38Packages.gtts</li>
<li>python38Packages.holoviews</li>
<li>python38Packages.huggingface-hub</li>
<li>python38Packages.hvplot</li>
<li>python38Packages.hyperopt</li>
<li>python38Packages.ignite</li>
<li>python38Packages.image-match</li>
<li>python38Packages.imagecorruptions</li>
<li>python38Packages.influxdb</li>
<li>python38Packages.influxgraph</li>
<li>python38Packages.intake</li>
<li>python38Packages.internetarchive</li>
<li>python38Packages.ipydatawidgets</li>
<li>python38Packages.labelbox</li>
<li>python38Packages.libagent</li>
<li>python38Packages.lsassy</li>
<li>python38Packages.mayavi</li>
<li>python38Packages.mesa</li>
<li>python38Packages.moviepy</li>
<li>python38Packages.mplfinance</li>
<li>python38Packages.msldap</li>
<li>python38Packages.nagiosplugin</li>
<li>python38Packages.nbsmoke</li>
<li>python38Packages.ndjson</li>
<li>python38Packages.nipype</li>
<li>python38Packages.nltk</li>
<li>python38Packages.nototools</li>
<li>python38Packages.osmpythontools</li>
<li>python38Packages.pandas</li>
<li>python38Packages.pandas-datareader</li>
<li>python38Packages.panel</li>
<li>python38Packages.papermill</li>
<li>python38Packages.parfive</li>
<li>python38Packages.partd</li>
<li>python38Packages.perfplot</li>
<li>python38Packages.phik</li>
<li>python38Packages.pims</li>
<li>python38Packages.plexapi</li>
<li>python38Packages.ppscore</li>
<li>python38Packages.proglog</li>
<li>python38Packages.pvlib</li>
<li>python38Packages.pyarrow</li>
<li>python38Packages.pybids</li>
<li>python38Packages.pyfftw</li>
<li>python38Packages.pygmt</li>
<li>python38Packages.pymatgen</li>
<li>python38Packages.pyppeteer</li>
<li>python38Packages.pypykatz</li>
<li>python38Packages.pyregion</li>
<li>python38Packages.pytest-astropy</li>
<li>python38Packages.pytest-astropy-header</li>
<li>python38Packages.python-daemon</li>
<li>python38Packages.python-miio</li>
<li>python38Packages.python-mpv-jsonipc</li>
<li>python38Packages.pytrends</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.radio_beam</li>
<li>python38Packages.rapidfuzz</li>
<li>python38Packages.rbtools</li>
<li>python38Packages.rdkit</li>
<li>python38Packages.reproject</li>
<li>python38Packages.rising</li>
<li>python38Packages.rpy2</li>
<li>python38Packages.sacremoses</li>
<li>python38Packages.salmon-mail</li>
<li>python38Packages.scikit-bio</li>
<li>python38Packages.scikitimage</li>
<li>python38Packages.seaborn</li>
<li>python38Packages.sfepy</li>
<li>python38Packages.slicedimage</li>
<li>python38Packages.slicer</li>
<li>python38Packages.spacy</li>
<li>python38Packages.sparse</li>
<li>python38Packages.spectral-cube</li>
<li>python38Packages.spyder-kernels_0_5</li>
<li>python38Packages.spyder_3</li>
<li>python38Packages.statsmodels</li>
<li>python38Packages.streamz</li>
<li>python38Packages.stumpy</li>
<li>python38Packages.stytra</li>
<li>python38Packages.sumo</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.tables</li>
<li>python38Packages.tensorly</li>
<li>python38Packages.test-tube</li>
<li>python38Packages.thinc</li>
<li>python38Packages.toggl-cli</li>
<li>python38Packages.tqdm</li>
<li>python38Packages.traittypes</li>
<li>trezor_agent (python38Packages.trezor_agent)</li>
<li>python38Packages.tubeup</li>
<li>python38Packages.uproot3</li>
<li>python38Packages.vega</li>
<li>python38Packages.vega_datasets</li>
<li>python38Packages.vidstab</li>
<li>python38Packages.wrf-python</li>
<li>python38Packages.xarray</li>
<li>python38Packages.yfinance</li>
<li>python39Packages.Quandl (python39Packages.quandl)</li>
<li>python39Packages.acoustics</li>
<li>python39Packages.afdko</li>
<li>python39Packages.aioinflux</li>
<li>python39Packages.aiosmb</li>
<li>python39Packages.altair</li>
<li>python39Packages.ansible-kernel</li>
<li>python39Packages.ansible-runner</li>
<li>python39Packages.aplpy</li>
<li>python39Packages.apptools</li>
<li>python39Packages.asdf</li>
<li>python39Packages.atomman</li>
<li>python39Packages.b2sdk</li>
<li>python39Packages.batchgenerators</li>
<li>python39Packages.bkcharts</li>
<li>python39Packages.caffe</li>
<li>python39Packages.catboost</li>
<li>python39Packages.cirq</li>
<li>python39Packages.csvs-to-sqlite</li>
<li>python39Packages.cufflinks</li>
<li>python39Packages.dask</li>
<li>python39Packages.dask-gateway</li>
<li>python39Packages.dask-image</li>
<li>python39Packages.dask-mpi</li>
<li>python39Packages.datasets</li>
<li>python39Packages.dipy</li>
<li>python39Packages.distributed</li>
<li>python39Packages.dnachisel</li>
<li>python39Packages.drms</li>
<li>python39Packages.duckdb</li>
<li>python39Packages.flammkuchen</li>
<li>python39Packages.gdown</li>
<li>python39Packages.genome-collector</li>
<li>python39Packages.glymur</li>
<li>python39Packages.gradient</li>
<li>python39Packages.gradient-utils</li>
<li>python39Packages.graphtage</li>
<li>python39Packages.gtts</li>
<li>python39Packages.holoviews</li>
<li>python39Packages.huggingface-hub</li>
<li>python39Packages.hvplot</li>
<li>python39Packages.hyperopt</li>
<li>python39Packages.ignite</li>
<li>python39Packages.image-match</li>
<li>python39Packages.imagecorruptions</li>
<li>python39Packages.imbalanced-learn</li>
<li>python39Packages.influxdb</li>
<li>python39Packages.influxgraph</li>
<li>python39Packages.intake</li>
<li>python39Packages.internetarchive</li>
<li>python39Packages.ipydatawidgets</li>
<li>python39Packages.kaggle</li>
<li>python39Packages.keepkey_agent</li>
<li>python39Packages.labelbox</li>
<li>python39Packages.ledger_agent</li>
<li>python39Packages.libagent</li>
<li>python39Packages.lsassy</li>
<li>python39Packages.magic-wormhole</li>
<li>python39Packages.mathlibtools</li>
<li>python39Packages.mayavi</li>
<li>python39Packages.mesa</li>
<li>python39Packages.moviepy</li>
<li>python39Packages.mplfinance</li>
<li>python39Packages.msldap</li>
<li>python39Packages.nagiosplugin</li>
<li>python39Packages.nbsmoke</li>
<li>python39Packages.ndjson</li>
<li>python39Packages.nipype</li>
<li>python39Packages.nltk</li>
<li>python39Packages.nototools</li>
<li>python39Packages.osmpythontools</li>
<li>python39Packages.pandas</li>
<li>python39Packages.pandas-datareader</li>
<li>python39Packages.panel</li>
<li>python39Packages.papermill</li>
<li>python39Packages.papis</li>
<li>python39Packages.parfive</li>
<li>python39Packages.partd</li>
<li>python39Packages.perfplot</li>
<li>python39Packages.pims</li>
<li>python39Packages.plexapi</li>
<li>python39Packages.ppscore</li>
<li>python39Packages.proglog</li>
<li>python39Packages.pvlib</li>
<li>python39Packages.pyarrow</li>
<li>python39Packages.pybids</li>
<li>python39Packages.pyfftw</li>
<li>python39Packages.pygmt</li>
<li>python39Packages.pymatgen</li>
<li>python39Packages.pyppeteer</li>
<li>python39Packages.pypykatz</li>
<li>python39Packages.pyregion</li>
<li>python39Packages.pytest-astropy</li>
<li>python39Packages.pytest-astropy-header</li>
<li>python39Packages.python-daemon</li>
<li>python39Packages.python-miio</li>
<li>python39Packages.python-mpv-jsonipc</li>
<li>python39Packages.pytrends</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.radio_beam</li>
<li>python39Packages.rapidfuzz</li>
<li>python39Packages.rbtools</li>
<li>python39Packages.rdkit</li>
<li>python39Packages.reproject</li>
<li>python39Packages.rising</li>
<li>python39Packages.rpy2</li>
<li>python39Packages.sacremoses</li>
<li>python39Packages.salmon-mail</li>
<li>python39Packages.scikit-bio</li>
<li>python39Packages.scikitimage</li>
<li>python39Packages.seaborn</li>
<li>python39Packages.sfepy</li>
<li>python39Packages.slicedimage</li>
<li>python39Packages.slicer</li>
<li>python39Packages.spacy</li>
<li>python39Packages.spectral-cube</li>
<li>python39Packages.spyder-kernels_0_5</li>
<li>python39Packages.spyder_3</li>
<li>python39Packages.statsmodels</li>
<li>python39Packages.streamz</li>
<li>python39Packages.sumo</li>
<li>python39Packages.sunpy</li>
<li>python39Packages.tables</li>
<li>python39Packages.test-tube</li>
<li>python39Packages.thinc</li>
<li>python39Packages.tqdm</li>
<li>python39Packages.traittypes</li>
<li>python39Packages.trezor_agent</li>
<li>python39Packages.tubeup</li>
<li>python39Packages.uproot3</li>
<li>python39Packages.vega</li>
<li>python39Packages.vega_datasets</li>
<li>python39Packages.vidstab</li>
<li>python39Packages.wrf-python</li>
<li>python39Packages.xarray</li>
<li>python39Packages.yfinance</li>
<li>r128gain</li>
<li>sacad</li>
<li>saga</li>
<li>spotdl</li>
<li>streamlit</li>
<li>tartube</li>
<li>tebreak</li>
<li>theharvester</li>
<li>tinyprog</li>
<li>truvari</li>
<li>twitter-color-emoji</li>
<li>urlwatch</li>
<li>vigra</li>
<li>visidata</li>
</ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118454` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
37 packages marked as broken and skipped:
</summary>
<ul>
<li>mlflow-server</li>
<li>python38Packages.arviz</li>
<li>python38Packages.clifford</li>
<li>python38Packages.dask-jobqueue</li>
<li>python38Packages.dask-xgboost</li>
<li>python38Packages.dftfit</li>
<li>python38Packages.geopandas</li>
<li>python38Packages.lammps-cython</li>
<li>python38Packages.mlflow</li>
<li>python38Packages.mlxtend</li>
<li>python38Packages.optuna</li>
<li>python38Packages.osmnx</li>
<li>python38Packages.pymatgen-lammps</li>
<li>python38Packages.pymc3</li>
<li>python38Packages.rl-coach</li>
<li>python38Packages.shap</li>
<li>python38Packages.skorch</li>
<li>python38Packages.textacy</li>
<li>python38Packages.tiledb</li>
<li>python38Packages.trackpy</li>
<li>python39Packages.dask-jobqueue</li>
<li>python39Packages.dask-xgboost</li>
<li>python39Packages.geopandas</li>
<li>python39Packages.google-cloud-automl</li>
<li>python39Packages.google-cloud-bigquery</li>
<li>python39Packages.google-cloud-monitoring</li>
<li>python39Packages.lammps-cython</li>
<li>python39Packages.mlflow</li>
<li>python39Packages.mlxtend</li>
<li>python39Packages.optuna</li>
<li>python39Packages.osmnx</li>
<li>python39Packages.pymatgen-lammps</li>
<li>python39Packages.rl-coach</li>
<li>python39Packages.skorch</li>
<li>python39Packages.textacy</li>
<li>python39Packages.tiledb</li>
<li>python39Packages.trackpy</li>
</ul>
</details>
<details>

  <summary>18 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>apache-airflow: log was empty</li>
<li>python38Packages.cnvkit: log was empty</li>
<li>python38Packages.dask-glm: log was empty</li>
<li>python38Packages.dask-ml: log was empty</li>
<li>python38Packages.imgaug: log was empty</li>
<li>python38Packages.nilearn: log was empty</li>
<li>python38Packages.pywick: log was empty</li>
<li>python38Packages.tablib: log was empty</li>
<li>python38Packages.transformers: log was empty</li>
<li>python38Packages.xgboost: log was empty</li>
<li>python39Packages.cnvkit: log was empty</li>
<li>python39Packages.dask-glm: log was empty</li>
<li>python39Packages.imgaug: log was empty</li>
<li>python39Packages.nilearn: log was empty</li>
<li>python39Packages.pywick: log was empty</li>
<li>python39Packages.tablib: log was empty</li>
<li>python39Packages.transformers: log was empty</li>
<li>python39Packages.xgboost: log was empty</li>
  </ul>
</details>
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python38Packages.spyder-kernels_0_5: log was empty</li>
<li>python38Packages.spyder_3: log was empty</li>
  </ul>
</details>
<details>
<summary>
336 packages built:
</summary>
<ul>
<li>aiodnsbrute</li>
<li>backblaze-b2</li>
<li>check_systemd</li>
<li>checkov</li>
<li>conan</li>
<li>cq-editor</li>
<li>csvs-to-sqlite (python38Packages.csvs-to-sqlite)</li>
<li>deja-dup</li>
<li>dpt-rp1-py</li>
<li>duplicity</li>
<li>duply</li>
<li>enblend-enfuse</li>
<li>evdevremapkeys</li>
<li>evillimiter</li>
<li>ffmpeg-normalize</li>
<li>gdown (python38Packages.gdown)</li>
<li>gnome-keysign</li>
<li>graph-cli</li>
<li>hdf5_1_10</li>
<li>home-assistant</li>
<li>hugin</li>
<li>jellyfin-mpv-shim</li>
<li>kaggle (python38Packages.kaggle)</li>
<li>keepkey_agent (python38Packages.keepkey_agent)</li>
<li>ldeep</li>
<li>ledger_agent (python38Packages.ledger_agent)</li>
<li>lieer</li>
<li>luigi</li>
<li>magic-wormhole (python38Packages.magic-wormhole)</li>
<li>manim</li>
<li>mathlibtools (python38Packages.mathlibtools)</li>
<li>mnemosyne</li>
<li>noto-fonts-emoji</li>
<li>nrfutil</li>
<li>ocrmypdf</li>
<li>papis (python38Packages.papis)</li>
<li>past-time</li>
<li>plex-mpv-shim</li>
<li>python38Packages.Quandl (python38Packages.quandl)</li>
<li>python38Packages.acoustics</li>
<li>python38Packages.afdko</li>
<li>python38Packages.aioinflux</li>
<li>python38Packages.aiosmb</li>
<li>python38Packages.altair</li>
<li>python38Packages.ansible-kernel</li>
<li>python38Packages.ansible-runner</li>
<li>python38Packages.aplpy</li>
<li>python38Packages.apptools</li>
<li>python38Packages.asdf</li>
<li>python38Packages.atomman</li>
<li>python38Packages.b2sdk</li>
<li>python38Packages.batchgenerators</li>
<li>python38Packages.bkcharts</li>
<li>python38Packages.caffe</li>
<li>python38Packages.catboost</li>
<li>python38Packages.cirq</li>
<li>python38Packages.cufflinks</li>
<li>python38Packages.dask</li>
<li>python38Packages.dask-gateway</li>
<li>python38Packages.dask-image</li>
<li>python38Packages.dask-mpi</li>
<li>python38Packages.datasets</li>
<li>python38Packages.datashader</li>
<li>python38Packages.dipy</li>
<li>python38Packages.distributed</li>
<li>python38Packages.dnachisel</li>
<li>python38Packages.drms</li>
<li>python38Packages.duckdb</li>
<li>python38Packages.fastparquet</li>
<li>python38Packages.flammkuchen</li>
<li>python38Packages.genome-collector</li>
<li>python38Packages.glymur</li>
<li>python38Packages.google-cloud-automl</li>
<li>python38Packages.google-cloud-bigquery</li>
<li>python38Packages.google-cloud-monitoring</li>
<li>python38Packages.gradient</li>
<li>python38Packages.gradient-utils</li>
<li>python38Packages.graphtage</li>
<li>python38Packages.graspologic</li>
<li>python38Packages.gtts</li>
<li>python38Packages.holoviews</li>
<li>python38Packages.huggingface-hub</li>
<li>python38Packages.hvplot</li>
<li>python38Packages.hyperopt</li>
<li>python38Packages.ignite</li>
<li>python38Packages.image-match</li>
<li>python38Packages.imagecorruptions</li>
<li>python38Packages.imbalanced-learn</li>
<li>python38Packages.influxdb</li>
<li>python38Packages.influxgraph</li>
<li>python38Packages.intake</li>
<li>python38Packages.internetarchive</li>
<li>python38Packages.ipydatawidgets</li>
<li>python38Packages.labelbox</li>
<li>python38Packages.libagent</li>
<li>python38Packages.lsassy</li>
<li>python38Packages.mayavi</li>
<li>python38Packages.mesa</li>
<li>python38Packages.moviepy</li>
<li>python38Packages.mplfinance</li>
<li>python38Packages.msldap</li>
<li>python38Packages.nagiosplugin</li>
<li>python38Packages.nbsmoke</li>
<li>python38Packages.ndjson</li>
<li>python38Packages.nipype</li>
<li>python38Packages.nltk</li>
<li>python38Packages.nototools</li>
<li>python38Packages.osmpythontools</li>
<li>python38Packages.pandas</li>
<li>python38Packages.pandas-datareader</li>
<li>python38Packages.panel</li>
<li>python38Packages.papermill</li>
<li>python38Packages.parfive</li>
<li>python38Packages.partd</li>
<li>python38Packages.perfplot</li>
<li>python38Packages.phik</li>
<li>python38Packages.pims</li>
<li>python38Packages.plexapi</li>
<li>python38Packages.ppscore</li>
<li>python38Packages.proglog</li>
<li>python38Packages.pvlib</li>
<li>python38Packages.pyarrow</li>
<li>python38Packages.pybids</li>
<li>python38Packages.pyfftw</li>
<li>python38Packages.pygmt</li>
<li>python38Packages.pymatgen</li>
<li>python38Packages.pyppeteer</li>
<li>python38Packages.pypykatz</li>
<li>python38Packages.pyregion</li>
<li>python38Packages.pytest-astropy</li>
<li>python38Packages.pytest-astropy-header</li>
<li>python38Packages.python-daemon</li>
<li>python38Packages.python-miio</li>
<li>python38Packages.python-mpv-jsonipc</li>
<li>python38Packages.pytrends</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.radio_beam</li>
<li>python38Packages.rapidfuzz</li>
<li>python38Packages.rbtools</li>
<li>python38Packages.rdkit</li>
<li>python38Packages.reproject</li>
<li>python38Packages.rising</li>
<li>python38Packages.rpy2</li>
<li>python38Packages.sacremoses</li>
<li>python38Packages.salmon-mail</li>
<li>python38Packages.scikit-bio</li>
<li>python38Packages.scikitimage</li>
<li>python38Packages.seaborn</li>
<li>python38Packages.sfepy</li>
<li>python38Packages.slicedimage</li>
<li>python38Packages.slicer</li>
<li>python38Packages.spacy</li>
<li>python38Packages.sparse</li>
<li>python38Packages.spectral-cube</li>
<li>python38Packages.statsmodels</li>
<li>python38Packages.streamz</li>
<li>python38Packages.stumpy</li>
<li>python38Packages.stytra</li>
<li>python38Packages.sumo</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.tables</li>
<li>python38Packages.tensorly</li>
<li>python38Packages.test-tube</li>
<li>python38Packages.thinc</li>
<li>python38Packages.toggl-cli</li>
<li>python38Packages.tqdm</li>
<li>python38Packages.traittypes</li>
<li>trezor_agent (python38Packages.trezor_agent)</li>
<li>python38Packages.tubeup</li>
<li>python38Packages.uproot3</li>
<li>python38Packages.vega</li>
<li>python38Packages.vega_datasets</li>
<li>python38Packages.vidstab</li>
<li>python38Packages.wrf-python</li>
<li>python38Packages.xarray</li>
<li>python38Packages.yfinance</li>
<li>python39Packages.Quandl (python39Packages.quandl)</li>
<li>python39Packages.acoustics</li>
<li>python39Packages.afdko</li>
<li>python39Packages.aioinflux</li>
<li>python39Packages.aiosmb</li>
<li>python39Packages.altair</li>
<li>python39Packages.ansible-kernel</li>
<li>python39Packages.ansible-runner</li>
<li>python39Packages.aplpy</li>
<li>python39Packages.apptools</li>
<li>python39Packages.asdf</li>
<li>python39Packages.atomman</li>
<li>python39Packages.b2sdk</li>
<li>python39Packages.batchgenerators</li>
<li>python39Packages.bkcharts</li>
<li>python39Packages.caffe</li>
<li>python39Packages.catboost</li>
<li>python39Packages.cirq</li>
<li>python39Packages.csvs-to-sqlite</li>
<li>python39Packages.cufflinks</li>
<li>python39Packages.dask</li>
<li>python39Packages.dask-gateway</li>
<li>python39Packages.dask-image</li>
<li>python39Packages.dask-mpi</li>
<li>python39Packages.datasets</li>
<li>python39Packages.dipy</li>
<li>python39Packages.distributed</li>
<li>python39Packages.dnachisel</li>
<li>python39Packages.drms</li>
<li>python39Packages.duckdb</li>
<li>python39Packages.flammkuchen</li>
<li>python39Packages.gdown</li>
<li>python39Packages.genome-collector</li>
<li>python39Packages.glymur</li>
<li>python39Packages.gradient</li>
<li>python39Packages.gradient-utils</li>
<li>python39Packages.graphtage</li>
<li>python39Packages.gtts</li>
<li>python39Packages.holoviews</li>
<li>python39Packages.huggingface-hub</li>
<li>python39Packages.hvplot</li>
<li>python39Packages.hyperopt</li>
<li>python39Packages.ignite</li>
<li>python39Packages.image-match</li>
<li>python39Packages.imagecorruptions</li>
<li>python39Packages.imbalanced-learn</li>
<li>python39Packages.influxdb</li>
<li>python39Packages.influxgraph</li>
<li>python39Packages.intake</li>
<li>python39Packages.internetarchive</li>
<li>python39Packages.ipydatawidgets</li>
<li>python39Packages.kaggle</li>
<li>python39Packages.keepkey_agent</li>
<li>python39Packages.labelbox</li>
<li>python39Packages.ledger_agent</li>
<li>python39Packages.libagent</li>
<li>python39Packages.lsassy</li>
<li>python39Packages.magic-wormhole</li>
<li>python39Packages.mathlibtools</li>
<li>python39Packages.mayavi</li>
<li>python39Packages.mesa</li>
<li>python39Packages.moviepy</li>
<li>python39Packages.mplfinance</li>
<li>python39Packages.msldap</li>
<li>python39Packages.nagiosplugin</li>
<li>python39Packages.nbsmoke</li>
<li>python39Packages.ndjson</li>
<li>python39Packages.nipype</li>
<li>python39Packages.nltk</li>
<li>python39Packages.nototools</li>
<li>python39Packages.osmpythontools</li>
<li>python39Packages.pandas</li>
<li>python39Packages.pandas-datareader</li>
<li>python39Packages.panel</li>
<li>python39Packages.papermill</li>
<li>python39Packages.papis</li>
<li>python39Packages.parfive</li>
<li>python39Packages.partd</li>
<li>python39Packages.perfplot</li>
<li>python39Packages.pims</li>
<li>python39Packages.plexapi</li>
<li>python39Packages.ppscore</li>
<li>python39Packages.proglog</li>
<li>python39Packages.pvlib</li>
<li>python39Packages.pyarrow</li>
<li>python39Packages.pybids</li>
<li>python39Packages.pyfftw</li>
<li>python39Packages.pygmt</li>
<li>python39Packages.pymatgen</li>
<li>python39Packages.pyppeteer</li>
<li>python39Packages.pypykatz</li>
<li>python39Packages.pyregion</li>
<li>python39Packages.pytest-astropy</li>
<li>python39Packages.pytest-astropy-header</li>
<li>python39Packages.python-daemon</li>
<li>python39Packages.python-miio</li>
<li>python39Packages.python-mpv-jsonipc</li>
<li>python39Packages.pytrends</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.radio_beam</li>
<li>python39Packages.rapidfuzz</li>
<li>python39Packages.rbtools</li>
<li>python39Packages.rdkit</li>
<li>python39Packages.reproject</li>
<li>python39Packages.rising</li>
<li>python39Packages.rpy2</li>
<li>python39Packages.sacremoses</li>
<li>python39Packages.salmon-mail</li>
<li>python39Packages.scikit-bio</li>
<li>python39Packages.scikitimage</li>
<li>python39Packages.seaborn</li>
<li>python39Packages.sfepy</li>
<li>python39Packages.slicedimage</li>
<li>python39Packages.slicer</li>
<li>python39Packages.spacy</li>
<li>python39Packages.spectral-cube</li>
<li>python39Packages.spyder-kernels_0_5</li>
<li>python39Packages.spyder_3</li>
<li>python39Packages.statsmodels</li>
<li>python39Packages.streamz</li>
<li>python39Packages.sumo</li>
<li>python39Packages.sunpy</li>
<li>python39Packages.tables</li>
<li>python39Packages.test-tube</li>
<li>python39Packages.thinc</li>
<li>python39Packages.tqdm</li>
<li>python39Packages.traittypes</li>
<li>python39Packages.trezor_agent</li>
<li>python39Packages.tubeup</li>
<li>python39Packages.uproot3</li>
<li>python39Packages.vega</li>
<li>python39Packages.vega_datasets</li>
<li>python39Packages.vidstab</li>
<li>python39Packages.wrf-python</li>
<li>python39Packages.xarray</li>
<li>python39Packages.yfinance</li>
<li>r128gain</li>
<li>sacad</li>
<li>saga</li>
<li>spotdl</li>
<li>streamlit</li>
<li>tartube</li>
<li>tebreak</li>
<li>theharvester</li>
<li>tinyprog</li>
<li>truvari</li>
<li>twitter-color-emoji</li>
<li>urlwatch</li>
<li>vigra</li>
<li>visidata</li>
</ul>
</details>
